File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_g2/structural_34_avl_tree.py
class Node_9:
    def __init__(self, v_9):
        self.v_9 = v_9
        self.r_9 = None
        self.l_9 = None
        self.height = 1

def h_9(node):
    if not node:
        return 0
    return node.height

def b_9(node):
    if not node:
        return 0
    return h_9(node.l_9) - h_9(node.r_9)

def rr_9(y):
    x = y.l_9
    sub = x.r_9
    x.r_9 = y
    y.l_9 = sub
    y.height = 1 + max(h_9(y.l_9), h_9(y.r_9))
    x.height = 1 + max(h_9(x.l_9), h_9(x.r_9))
    return x

def rl_9(x):
    y = x.r_9
    sub = y.l_9
    y.l_9 = x
    x.r_9 = sub
    x.height = 1 + max(h_9(x.l_9), h_9(x.r_9))
    y.height = 1 + max(h_9(y.l_9), h_9(y.r_9))
    return y

def ins_9(rt_9, v_9):
    if not rt_9:
        return Node_9(v_9)
    if v_9 < rt_9.v_9:
        rt_9.l_9 = ins_9(rt_9.l_9, v_9)
    elif v_9 > rt_9.v_9:
        rt_9.r_9 = ins_9(rt_9.r_9, v_9)
    else:
        return rt_9

    rt_9.height = 1 + max(h_9(rt_9.l_9), h_9(rt_9.r_9))
    bf_9 = b_9(rt_9)

    if bf_9 > 1 and v_9 < rt_9.l_9.v_9:
        return rr_9(rt_9)
    if bf_9 < -1 and v_9 > rt_9.r_9.v_9:
        return rl_9(rt_9)
    if bf_9 > 1 and v_9 > rt_9.l_9.v_9:
        rt_9.l_9 = rl_9(rt_9.l_9)
        return rr_9(rt_9)
    if bf_9 < -1 and v_9 < rt_9.r_9.v_9:
        rt_9.r_9 = rr_9(rt_9.r_9)
        return rl_9(rt_9)

    return rt_9

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_g2/structural_34_avl_tree.java
class Node_9 {
    int k_9, h_9;
    Node_9 left, right;
    Node_9(int d) {
        k_9 = d;
        h_9 = 1;
    }
}
public class Tree_9 {
    public static void main(String[] args) {
        Tree_9 t = new Tree_9();
        Node_9 r = null;
        r = t.ins_9(r, 10);
    }
    int max(int a, int b) {
        return (a > b) ? a : b;
    }
    int h_9(Node_9 N) {
        if (N == null) return 0;
        return N.h_9;
    }
    int gb_9(Node_9 N) {
        if (N == null) return 0;
        return h_9(N.left) - h_9(N.right);
    }
    Node_9 upd_9(Node_9 n_9) {
        n_9.h_9 = 1 + max(h_9(n_9.left), h_9(n_9.right));
        return n_9;
    }
    Node_9 rr_9(Node_9 y) {
        Node_9 x = y.left;
        Node_9 T2 = x.right;
        x.right = y;
        y.left = T2;
        upd_9(y);
        upd_9(x);
        return x;
    }
    Node_9 rl_9(Node_9 x) {
        Node_9 y = x.right;
        Node_9 T2 = y.left;
        y.left = x;
        x.right = T2;
        upd_9(x);
        upd_9(y);
        return y;
    }
    Node_9 ins_9(Node_9 n_9, int k_9) {
        if (n_9 == null) return (new Node_9(k_9));
        if (k_9 < n_9.k_9) n_9.left = ins_9(n_9.left, k_9);
        else if (k_9 > n_9.k_9) n_9.right = ins_9(n_9.right, k_9);
        else return n_9;
        upd_9(n_9);
        int b_9 = gb_9(n_9);
        if (b_9 > 1 && k_9 < n_9.left.k_9) return rr_9(n_9);
        if (b_9 < -1 && k_9 > n_9.right.k_9) return rl_9(n_9);
        if (b_9 > 1 && k_9 > n_9.left.k_9) {
            n_9.left = rl_9(n_9.left);
            return rr_9(n_9);
        }
        if (b_9 < -1 && k_9 < n_9.right.k_9) {
            n_9.right = rr_9(n_9.right);
            return rl_9(n_9);
        }
        return n_9;
    }
}